Modules/Filtering/LabelMap/test/itkStatisticsLabelMapFilterGTest.cxx

```diff
@@ -0,0 +1,251 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkGTest.h"
+
+#include "itkImage.h"
+#include "itkLabelImageToStatisticsLabelMapFilter.h"
+#include "itkImageRegionIterator.h"
+#include <algorithm>
+
+
+namespace
+{
+
+class StatisticsLabelMapFixture : public ::testing::Test
+{
+public:
+  StatisticsLabelMapFixture() = default;
+  ~StatisticsLabelMapFixture() override = default;
+
+protected:
+  void
+  SetUp() override
+  {}
+  void
+  TearDown() override
+  {}
+
+  template <unsigned int D, typename TPixelType = unsigned short>
+  struct FixtureUtilities
+  {
+    static const unsigned int Dimension = D;
+
+    using PixelType = TPixelType;
+    using ImageType = itk::Image<PixelType, Dimension>;
+
+    using LabelPixelType = unsigned char;
+    using LabelImageType = itk::Image<LabelPixelType, Dimension>;
+
+    using LabelObjectType = itk::StatisticsLabelObject<LabelPixelType, Dimension>;
+    using StatisticsLabelMapType = itk::LabelMap<LabelObjectType>;
+
+
+    static typename ImageType::Pointer
+    CreateImage()
+    {
+      typename ImageType::Pointer image = ImageType::New();
+
+      typename ImageType::SizeType imageSize;
+      imageSize.Fill(25);
+      image->SetRegions(typename ImageType::RegionType(imageSize));
+      image->Allocate();
+      image->FillBuffer(0);
+
+      return image;
+    }
+
+    static typename ImageType::Pointer
+    CreateImageRandom(PixelType randMax = 500, unsigned int randSeed = 0)
+    {
+      typename ImageType::Pointer image = ImageType::New();
+
+      typename ImageType::SizeType imageSize;
+      imageSize.Fill(25);
+      image->SetRegions(typename ImageType::RegionType(imageSize));
+      image->Allocate();
+      image->FillBuffer(0);
+
+      srand(randSeed);
+
+      itk::ImageRegionIterator<ImageType> it(image, image->GetLargestPossibleRegion());
+      while (!it.IsAtEnd())
+      {
+        it.Set(rand() % randMax);
+        ++it;
+      }
+      return image;
+    }
+
+
+    static typename LabelImageType::Pointer
+    CreateLabelImage()
```

Comment:
Why not make CreateImage templated, and pass image type?

Reply:
The outer support class FixtureUtilities is templated so that they can share the same typedefs.